Source/Scene/Cesium3DTile.js

```diff
@@ -0,0 +1,1018 @@
+         *
+         * @readonly
+         *
+         * @private
+         */
+        commandsLength : {
+            get : function() {
+                return this._commandsLength;
+            }
+        }
+    });
+
+    var scratchJulianDate = new JulianDate();
+
+    /**
+     * Update whether the tile has expired.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.updateExpiration = function() {
+        if (defined(this.expireDate) && this.contentReady && !this.hasEmptyContent) {
+            var now = JulianDate.now(scratchJulianDate);
+            if (JulianDate.lessThan(this.expireDate, now)) {
+                this._contentState = Cesium3DTileContentState.EXPIRED;
+                this._expiredContent = this._content;
+            }
+        }
+    };
+
+    function updateExpireDate(tile) {
+        if (defined(tile.expireDuration)) {
+            var expireDurationDate = JulianDate.now(scratchJulianDate);
+            JulianDate.addSeconds(expireDurationDate, tile.expireDuration, expireDurationDate);
+
+            if (defined(tile.expireDate)) {
+                if (JulianDate.lessThan(tile.expireDate, expireDurationDate)) {
+                    JulianDate.clone(expireDurationDate, tile.expireDate);
+                }
+            } else {
+                tile.expireDate = JulianDate.clone(expireDurationDate);
+            }
+        }
+    }
+
+    /**
+     * Requests the tile's content.
+     * <p>
+     * The request may not be made if the Cesium Request Scheduler can't prioritize it.
+     * </p>
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.requestContent = function() {
+        var that = this;
+
+        if (this.hasEmptyContent) {
+            return false;
+        }
+
+        if (!this._requestServer.hasAvailableRequests()) {
+            return false;
+        }
+
+        var url = this._contentUrl;
+        if (defined(this.expireDate)) {
+            // Append a query parameter of the tile expiration date to prevent caching
+            var timestampQuery = '?expired=' + this.expireDate.toString();
+            url = joinUrls(url, timestampQuery, false);
+        }
+
+        var distance = this._distanceToCamera;
+        var promise = RequestScheduler.schedule(new Request({
+            url : url,
+            server : this._requestServer,
+            requestFunction : loadArrayBuffer,
+            type : RequestType.TILES3D,
+            distance : distance
+        }));
+
+        if (!defined(promise)) {
+            return false;
+        }
+
+        this._contentState = Cesium3DTileContentState.LOADING;
+        this._contentReadyToProcessPromise = when.defer();
+        this._contentReadyPromise = when.defer();
+
+        promise.then(function(arrayBuffer) {
+            if (that.isDestroyed()) {
+                // Tile is unloaded before the content finishes loading
+                return when.reject('tile is destroyed');
```

Comment:
Using `reject` for non-error situations is not an ideal architecture.  Is that what's going on here?  I assume this shouldn't be a `RuntimeError`?  We should consider resolving to undefined and have the dependant code check undefined instead of relying on rejection.